src/app/surveys/surveys.component.ts

```diff
@@ -72,7 +72,30 @@ export class SurveysComponent implements OnInit, AfterViewInit, OnDestroy {
 
   getSubmissions() {
     // get the no of submisson for each test from submisson table
-    return this.couchService.findAll('submissions', { 'selector': { 'type': 'survey' } });
+    return this.couchService.findAll('submissions', { 'selector': {
```

Comment:
Instead of using the long find selector, lets do 
```
this.couchService.findAll('submissions').pipe((submissions) => ...
```
and then filter the submissions with a javascript filter function.  That will be more readable, and I don't think this selector will perform well on the RasPi.